packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -445,8 +454,33 @@ def _deserialize(deserializer: typing.Optional[typing.Callable[[typing.Any], typ
     try:
         if value is None:
             return None
+        if deserializer == typing.Any:
```

Comment:
Is there a way we can resuse the `_get_deserializable_callable` etc code? https://github.com/Azure/autorest.python/blob/5730d9213ed9d455473ad49597ac01b297f949fc/packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2#L532

It seems to be quite similar, we just want to appply it to, say a dictionary of models

Reply:
yes, @iscai-msft , I refactor the vendored code in  [20221012-dictionary](https://github.com/Azure/autorest.python/pull/1520/commits/7946822d3f2382cfdb5eaad24168583594029210). Mainly for:
1. move logic of `_RestField._get_deserialize_callable_from_annotation()` to a global function `get_deserialize_callable_from_annotation`
2.  add logic to parse `Union` annotation in `get_deserialize_callable_from_annotation` since they are used for enum deserialization.